database/src/test/scala/za/co/absa/atum/database/runs/CreatePartitioningIfNotExists.scala

```diff
@@ -0,0 +1,192 @@
+      .execute { queryResult =>
+        assert(queryResult.hasNext)
+        val row = queryResult.next()
+        assert(row.getInt("status").contains(11))
+        assert(row.getString("status_text").contains("Partitioning created"))
+        row.getLong("id_partitioning").get
+      }
+
+    table("runs.partitionings").where(add("id_partitioning", partitioningID)) {partitioningResult =>
+      val row = partitioningResult.next()
+      // assert(row.getJsonB("partitioning").contains(partitioning)) TODO keys are reordered in JsonB and whitespaces removed
+      assert(row.getString("created_by").contains("Fantômas"))
+      assert(row.getOffsetDateTime("created_at").contains(now()))
+    }
+
+    val idFlow = table("flows.partitioning_to_flow").where(add("fk_partitioning", partitioningID)) { partToFlowResult =>
+      assert(partToFlowResult.hasNext)
+      val partToFlowRow = partToFlowResult.next()
+      val result = partToFlowRow.getLong("fk_flow")
+      assert(partToFlowRow.getString("created_by").contains("Fantômas"))
+      assert(!partToFlowResult.hasNext)
+      result.get
+    }
+
+    table("flows.flows").where(add("id_flow", idFlow)) {flowsResult =>
+      assert(flowsResult.hasNext)
+      val flowRow = flowsResult.next()
+      assert(flowRow.getString("flow_name").exists(_.startsWith("Custom flow #")))
+      assert(flowRow.getString("flow_description").contains(""))
+      assert(flowRow.getBoolean("from_pattern").contains(false))
+      assert(flowRow.getString("created_by").contains("Fantômas"))
+      assert(flowRow.getOffsetDateTime("created_at").contains(now()))
+      assert(!flowsResult.hasNext)
+    }
+  }
+  test("Partitioning created with parent partitioning that already exists") {
+    val parentPartitioningID = function(fncCreatePartitioningIfNotExists)
+      .setParam("i_partitioning", parentPartitioning)
+      .setParam("i_by_user", "Albert Einstein")
+      .execute { queryResult =>
+        assert(queryResult.hasNext)
+        val row = queryResult.next()
+        assert(row.getInt("status").contains(11))
+        assert(row.getString("status_text").contains("Partitioning created"))
+        row.getLong("id_partitioning").get
+      }
+
+    assert(
+      table("flows.partitioning_to_flow").count(add("fk_partitioning", parentPartitioningID)) == 1
+    )
+    val partitioningID = function(fncCreatePartitioningIfNotExists)
+      .setParam("i_partitioning", partitioning)
+      .setParam("i_by_user", "Fantômas")
+      .setParam("i_parent_partitioning", parentPartitioning)
+      .execute { queryResult =>
+        assert(queryResult.hasNext)
+        val row = queryResult.next()
+        assert(row.getInt("status").contains(11))
+        assert(row.getString("status_text").contains("Partitioning created"))
+        row.getLong("id_partitioning").get
+      }
+
+    assert(
+      table("flows.partitioning_to_flow").count(add("fk_partitioning", parentPartitioningID)) == 1
```

Comment:
(balta related, not AS) this `add` here really smells to me  but I technically understand why it is here; but still, the 'API' seems suboptimal on the first sight :D